### PR TITLE
<Popover/> - add support for `flip` and update `moveBy` properly

### PR DESCRIPTION
**Pull request by yardnsm**
This PR adds a `flip` prop in order to control Popper.JS's `flip` behaviour (more info can be found [here](https://popper.js.org/popper-documentation.html#modifiers..flip)).

In addition, the `moveBy` behaviour has been changed so when its enabled, the `flip` behaviour will be disabled in order to support passing negative values to the Content element will overlap the Target element.

Also, it fixes the misleading usage at per https://github.com/wix/wix-style-react/issues/2636.

----

**Update:** a new `fixed` prop was added. When set, it'll disable the `preventOverflow` modifier so the popover will stay at the same position at all times. Check the examples for more info.

### CODE REVIEW

File: packages/wix-ui-core/src/components/popover/Popover.tsx [hunk 1]
```diff
@@ -54,6 +54,16 @@ export interface PopoverProps {
   onKeyDown?: React.KeyboardEventHandler<HTMLDivElement>;
   /** Show show arrow from the content */
   showArrow?: boolean;
+  /**
+   * Whether to enable the flip behaviour. This behaviour used to flip the `<Popover/>`'s placement
```
Comment (hepiyellow) [inline packages/wix-ui-core/src/components/popover/Popover.tsx]:
```suggestion
   * Whether to enable the flip behaviour. This behaviour is used to flip the `<Popover/>`'s placement
```

File: packages/wix-ui-core/src/components/popover/Popover.tsx [hunk 2]
```diff
@@ -54,6 +54,16 @@ export interface PopoverProps {
...
+   * when it starts to overlap the target element (`<Popover.Element/>`).
+   */
+  flip?: boolean;
+  /**
+   * Whether to enable the fixed behaviour. This behaviour used to keep the `<Popover/>` at it's
```
Comment (hepiyellow) [inline packages/wix-ui-core/src/components/popover/Popover.tsx]:
```suggestion
   * Whether to enable the fixed behaviour. This behaviour is used to keep the `<Popover/>` at it's
```

**Comment by yardnsm** [comment]
@hepiyellow there isn't a way to test that so much, as this logic is specific to PopperJS.

https://github.com/wix/wix-style-react/pull/2676 does have some e2e tests for it.

File: packages/wix-ui-core/src/components/popover/Popover.tsx [hunk 3]
```diff
@@ -106,13 +116,38 @@ const getArrowShift = (shift: number | undefined, direction: string) => {
...
       gpuAcceleration: !shouldAnimate
+    },
+    flip: {
+      enabled: typeof moveBy === 'undefined' ? flip : !moveBy
+    },
+    preventOverflow: {
+      enabled: !fixed,
+    },
+    hide: {
+      enabled: !fixed
```
[the remark this answers is not in the document]
Comment (yardnsm) [reply packages/wix-ui-core/src/components/popover/Popover.tsx]: We'll want to use `hide` anyway in `preventOverflow` is `true`.

**Pull request merged**